### ISSUE TITLE
Question: is there any way of calling nox API inside a test framework (e.g. pytest)?

### ISSUE THREAD

**Issue by abravalheri** [theacodes/nox#333]
Thank you very much for nox. It is really handy and the API is very slick!
So much so that I was wondering if there is any way of calling it programatically from inside a test file to write integration/acceptance/end-to-end tests...

Let me explain a little bit more, it is very common for me to create CLI python programs. Therefore it is important to test them running using different set of (optional) dependencies; if they work correctly when installed via conda, venv, virtualenv, wheel, sdist, ...; if they are correctly build/packed using setup tools...
It looks to me that nox's session API (or very much all nox API) would be perfect for this use case.
However, it is nice to be able to use something like pytest to orchestrate and report these tests instead of directly creating nox tasks (after all, pytest allows debugging when things go wrong, we can use xdist, and loads of other niceties).

Optimistically, the flow `nox (as task manger) > pytest > nox python API (to create venvs, install packages and run commands inside tests)`  would be the ultimate integration/acceptance/... test tool.

So, to summarise: is there any way of using nox session's API from a regular python file? (and: is there a way of capturing `session.run`'s `stdout/stderr`?)

**Comment by theacodes** [comment]
Nox doesn't have a publicly documented API for this sort of thing, and I think it's probably outside of our use case. I would think that Invoke would be a better fit here.

I'm going to close this, but feel free to as further questions!

**Comment by abravalheri** [comment]
Thank you very much for the suggestion!

I will try to have a look on Invoke and probably borrow some code from nox for the virtualenv management.
